AsyncDisplayKit/ASRunLoopQueue.mm

```diff
@@ -48,9 +50,22 @@ + (instancetype)sharedDeallocationQueue
 
 - (void)releaseObjectInBackground:(id)object
 {
-  _queueLock.lock();
-  _queue.push_back(object);
-  _queueLock.unlock();
+  __weak __typeof__(self) weakSelf = self;
+  ASPerformBlockOnMainThread(^{
+    // Some associated objects cannot be deallocated in background (#2767). Always remove them on main instead.
+    objc_removeAssociatedObjects(object);
+    
+    __typeof__(self) strongSelf = weakSelf;
+    if (strongSelf) {
```

Comment:
Since this is such a hot code path, and since in practice the queue is going to be long-lived, let's avoid hitting the weak table here and just retain `self` to reduce contention.